### PR TITLE
FP55 - Basic Prompt Enginerring

### PR DESCRIPTION
**Pull request by TomerMiz10**
## Task link from JIRA:
https://mssplinter10-1711638164183.atlassian.net/jira/software/projects/FP/boards/6?selectedIssue=FP-55

## Description:
- chore: Update pull request template with new JIRA link and code owners (6c5828f)
- chore: Update .gitignore and pull request template (5ac95ee)
- chore: Improve user prompt in `constants.ts`. (ea8004d)
- chore: Update pull request template with new JIRA link and code owners (5a34039)
- chore: Update app.listen port to 3000 (ff07f3c)
- Refactor: Changed GuideResponseDTO to use an array of strings instead of a single step (4454398)
- Feat: Update OpenAIService to use different prompts and improve conversation formatting. (c211e08)
- Feat: Update OpenAIService to include user prompt, output indicator, and context guide (089fe70)
- chore: Update return type of generateGuide method (b47e353)
- Feat: Update example conversation in `constants.ts` and improve prompts like system ,user, output indicator and context. (711ddf8)
- Feat: Update System Prompt in `constants.ts` to return as un onordered list of HTML. (0827475)
- Chore: Edit System Prompt in `constants.ts` (04740bb)

## Code Owners:
@TomerMiz10 @JosephOri 

### Summary
_Write an overview_

### CODE REVIEW

File: src/openai/constants/constants.ts [hunk 1]
```diff
@@ -1,12 +1,33 @@
 import * as fs from 'fs';
-// export type ConversationMessages = Record<string, string>;
-// change back to this type, infer the conversation from backend and format into objects of messages to the api
-const messages = fs.readFileSync('src/openai/lib/data/messages.txt', 'utf8');
-export const MESSAGES: string = messages;
+const exampleConversation = fs.readFileSync('src/openai/lib/data/messages.txt', 'utf8');
```
Comment (JosephOri) [inline src/openai/constants/constants.ts]: change 'src/openai/lib/data/messages.txt' url to const and also 'utf8' to a const

File: src/openai/services/openai.service.ts [hunk 2]
```diff
@@ -2,57 +2,59 @@ import { Injectable, Logger  } from "@nestjs/common";
...
         this.openai = new OpenAIApi({ apiKey: this.configService.get('OPENAI_API_KEY') });
-        this.conversationMessages = MESSAGES;
     }
 
 
     formatConversation (chatConversation: string): any[] {
-        const messages = chatConversation;
-        console.log("messages: ", messages);
+        const messages = chatConversation.trim();
+        // Logger.log("messages: ", messages);
```
Comment (JosephOri) [inline src/openai/services/openai.service.ts]: remove or add commented code

File: src/openai/services/openai.service.ts [hunk 3]
```diff
@@ -2,57 +2,59 @@ import { Injectable, Logger  } from "@nestjs/common";
...
-            role: "user",
-            content: chatConversation
-        }];
+        const lines = messages.split('\n');
+        const formattedMessages = lines.map(line => {
+            const [role, ...messageParts] = line.replace('\r', '').split(': ');
+            const message = messageParts.join(': ');
+            return {name: role, role: role, content: message};
+        })
+        .filter(message => ['system', 'assistant', 'user', 'function'].includes(message.role) && message.content.trim() !== '');
```
Comment (JosephOri) [inline src/openai/services/openai.service.ts]: put this array into a const in constants file

**Review by JosephOri — CHANGES_REQUESTED** [review]

File: src/openai/constants/constants.ts (hunk 1 above)
Comment (TomerMiz10) [reply src/openai/constants/constants.ts]: done

File: src/openai/services/openai.service.ts (hunk 2 above)
Comment (TomerMiz10) [reply src/openai/services/openai.service.ts]: done.

File: src/openai/services/openai.service.ts (hunk 3 above)
Comment (TomerMiz10) [reply src/openai/services/openai.service.ts]: done

**Review by JosephOri — APPROVED** [review]

**Pull request merged**